src/main/java/at/codersbay/datastructures/Fractional.java

```diff
@@ -5,70 +5,133 @@
+				== this.getDenominator()) {
+			return true;
+		}
+		/*Numerator is 0*/
+		if(fractionalObject.getNumerator() == 0 
+				&& this.getNumerator() == 0) {
+			return true;
+		}
+		/*Numerators are equal after multiplication with other
+		 * fractionals denominator*/
+		if(this.getNumerator()*fractionalObject.getDenominator() 
+			== fractionalObject.getNumerator()*this.getDenominator()) 
+		{
+			return true;
+		}
+		
+
+		return false;
+	}
 
     public Fractional multiply(Fractional other) {
-        // TODO implement to fix tests in FractionalTest4
-        return new Fractional(1, 1);
+    	Integer calcDenom = 0;
+    	Integer calcNum = 0;
+    	calcDenom = this.getDenominator()*other.getDenominator();
+    	calcNum = this.getNumerator()*other.getNumerator();
+    	
+        return new Fractional(calcNum, calcDenom);
     }
 
     public Fractional multiply(Integer other) {
-        // TODO implement to fix tests in FractionalTest4
-        return new Fractional(1, 1);
+        return new Fractional((this.getNumerator()*other),(this.getDenominator()));
     }
 
     public Fractional divide(Fractional other) {
-        // TODO implement to fix tests in FractionalTest5
-        return new Fractional(1, 1);
+    	/*division by multiplication with the invers*/
+    	return this.multiply(this.invertedFractional(other));
     }
 
     public Fractional divide(Integer other) {
-        // TODO implement to fix tests in FractionalTest5
-        return new Fractional(1, 1);
+    	if (other == null || other == 0) {
+    		throw new IllegalArgumentException();
+    	} 
+    	return this.divide(new Fractional(other,1));
     }
 
     public Fractional add(Fractional other) {
-        // TODO implement to fix tests in FractionalTest6
-        return new Fractional(1, 1);
+    	int newDenom = 0;
+    	int newNum = 0;
+    	newNum = this.getNumerator()*other.getDenominator()+
+    			other.getNumerator()*this.getDenominator();
+    	newDenom = this.getDenominator()*other.getDenominator();
+        return new Fractional(newNum,newDenom);
     }
 
     public Fractional add(Integer other) {
-        // TODO implement to fix tests in FractionalTest6
-        return new Fractional(1, 1);
+    	if (other == null) {
+    		throw new IllegalArgumentException();
+    	}
+        return this.add(new Fractional(other, 1));
     }
 
     public Fractional subtract(Fractional other) {
-        // TODO implement to fix tests in FractionalTest7
-        return new Fractional(1, 1);
+     	int newDenom = 0;
+    	int newNum = 0;
+    	newNum = this.getNumerator()*other.getDenominator()-
+    			other.getNumerator()*this.getDenominator();
+    	newDenom = this.getDenominator()*other.getDenominator();
+        return new Fractional(newNum,newDenom);
     }
 
     public Fractional subtract(Integer other) {
-        // TODO implement to fix tests in FractionalTest7
-        return new Fractional(1, 1);
+        return this.subtract(new Fractional(other,1));
```

Comment:
really nice especially the reuse of methods in other methods! 👍

Reply:
thank you